src/extract.rs

```diff
@@ -13,6 +14,27 @@ pub(crate) enum FileType {
     Plaintext,
 }
 
+impl Default for FileType {
+    fn default() -> Self {
+        Self::Plaintext
+    }
+}
+
+impl<P: AsRef<Path>> From<P> for FileType {
+    /// Detect if the given path points to a Markdown, HTML, or plaintext file.
+    fn from(p: P) -> FileType {
+        let path = p.as_ref();
+        match path.extension() {
+            Some(ext) => match ext {
+                _ if ext == "md" => FileType::Markdown,
+                _ if (ext == "htm" || ext == "html") => FileType::HTML,
+                _ => FileType::Plaintext,
```

Comment:
We discussed whether we should `unwrap`, use `TryFrom` etc. Turns out, `OsStr` implements `PartialEq` for `&str`, so... I think we have a winner here :wink:

Reply:
Rust is awesome.

Reply:
and kudos for finding that